classes/02class/exercises/c02-network07/gepifanio/README.md

```diff
@@ -0,0 +1,191 @@
+--nat-gateway-id nat-0fbe2a7f66cdb2924
+
+An error occurred (MissingParameter) when calling the CreateRoute operation: The request must contain exactly one of gatewayId, natGatewayId, networkInterfaceId, vpcPeeringConnectionId, egressOnlyInternetGatewayId, transitGatewayId, localGatewayId or instanceId
+
+I have deleted my NAT gateway which caused this problem.
+
+```
+
+- Answer the following questions:
+  - How did you accomplish both instructions?
+    ```
+    Looking in the aws documentation about the commands, how to create a routing table, using the tags from the previous exercises to add name in the resources, and searching how to create route into a route table.
+    ```
+
+  - Why did you add 0.0.0.0 route to the IGW on public route table?
+    ```
+    To allow all ip address of the subnets and instances to access the internet
+
+    ```
+
+  - Why did you add 0.0.0.0 route to the NGW on private route table?
+    ```
+
+    To enable the private network to access the internet using public IP address that is attached to the NAT gateway.
+
+
+    ```
+    
+  - What is the difference of IGW and NGW?
+    ```
+
+    An Internet Gateway (IGW) is a logical connection between an Amazon VPC and the Internet. It is not a physical device. Only one can be associated with each VPC. It does not limit the bandwidth of Internet connectivity. (The only limitation on bandwidth is the size of the Amazon EC2 instance, and it applies to all traffic — internal to the VPC and out to the Internet.)
+    If a VPC does not have an Internet Gateway, then the resources in the VPC cannot be accessed from the Internet (unless the traffic flows via a corporate network and VPN/Direct Connect).
+    An Internet Gateway allows resources within your VPC to access the internet, and vice versa. In order for this to happen, there needs to be a routing table entry allowing a subnet to access the IGW.
+    That is to say — an IGW allows resources within your public subnet to access the internet, and the internet to access said resources.
+    A subnet is deemed to be a Public Subnet if it has a Route Table that directs traffic to the Internet Gateway.
+
+    A NAT Gateway does something similar, but with two main differences:
+    It allows resources in a private subnet to access the internet (think yum updates, external database connections, wget calls, OS patch, etc)
+    It only works one way. The internet at large cannot get through your NAT to your private resources unless you explicitly allow it.
+
+    ```
+    
+  - Can you delete the destination route to your VPC network? Why?
+    ```
+    No, because the subnets and the routes are all going through the VPC, and if I delete it, is not gonna work.
+    
+    ```
+    
+  - What happens if you do not associate your route table with any subnets?
+    ```
+    The subnets will not be able to communicate with the IGW and NGW for instance in our exercise, only between themselves. 
```

Comment:
Just a quick note here, in that case **the main (default) route table** will be associated and it **may or may not have** a route entry to an IGW/NGW.

More info here: https://docs.aws.amazon.com/vpc/latest/userguide/VPC_Route_Tables.html#RouteTableDetails